stablehlo/tests/interpret_iota.mlir

```diff
@@ -0,0 +1,1213 @@
+// RUN: stablehlo-interpreter --interpret -split-input-file %s | FileCheck %s
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si4_dim_0
+func.func @iota_op_test_si4_dim_0() -> tensor<3x4xi4> {
+  %0 = stablehlo.iota dim = 0 : tensor<3x4xi4>
+  func.return %0 : tensor<3x4xi4>
+  // CHECK-NEXT: tensor<3x4xi4>
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 2 : i4
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si4_dim_1
+func.func @iota_op_test_si4_dim_1() -> tensor<3x4xi4> {
+  %0 = stablehlo.iota dim = 1 : tensor<3x4xi4>
+  func.return %0 : tensor<3x4xi4>
+  // CHECK-NEXT: tensor<3x4xi4>
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 3 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 3 : i4
+  // CHECK-NEXT: 0 : i4
+  // CHECK-NEXT: 1 : i4
+  // CHECK-NEXT: 2 : i4
+  // CHECK-NEXT: 3 : i4
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si4_dim_0_overflow
```

Comment:
We should drop tests exercising the implementation defined behavior. This is what we followed for other ops like AddOp.